File: ats-ness-back-prod/src/main/java/nesstechnologies/fr/applicationesnbackend/entities/Referentiel.java
package nesstechnologies.fr.applicationesnbackend.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
@ToString(exclude = {"profilReferentiels", "candidat", "besoin"})
public class Referentiel implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @ElementCollection
    private  List<String> segments;
    @ElementCollection
    private List<String> sousSegments;
    @ElementCollection
    private List<String> technologie;

    @OneToMany(mappedBy = "referentiel", cascade = CascadeType.ALL,orphanRemoval = true)
    private List<ProfilReferentiel> profilReferentiels;

    @JsonIgnore
    @ManyToOne
    private Candidat candidat;

    @OneToOne(mappedBy = "referentiel")
    private Besoins besoin;



}

File: ats-ness-front-prod/src/app/demo/models/Referentiel.ts
import {Candidat} from "./condidat";
import {Besoin} from "./besoin";
import {ProfilReferentiel} from "./ProfilReferentiel";

export class Referentiel {
    id: number | undefined;
    segments: string[] = [];
    sousSegments: string[] = [];
    technologie: string[] = [];

    // Relation avec l'entité Candidat
    candidat: Candidat | undefined;
    besoin?: Besoin | undefined;
    profilReferentiels: ProfilReferentiel[] = [];

}
